src/components/CommentCard.jsx

```diff
@@ -1,6 +1,24 @@
+import { useContext, useState } from "react";
 import { FancyBox } from "./FancyBox";
+import { UserContext } from "./contexts/User";
+import { deleteComment } from "./api";
+import { ErrorComponent } from "./ErrorComponent";
 
-export function CommentCard({comment}) {
+export function CommentCard({comment, comments, setComments}) {
+    const {user} = useContext(UserContext)
+    const [isDeleting, setIsDeleting] = useState(false)
+    const [error, setError] = useState(null)
+
+    function removeComment() {
+        setIsDeleting(true)
+        deleteComment(comment.comment_id)
+        .then(() => {
+            
+            setIsDeleting(false)
+            setComments(comments.filter(notDeletedComment => notDeletedComment.comment_id !== comment.comment_id))
+        })
+        .catch(err => setError(err))
```

Comment:
Instead of catching and setting the error to be the axios error here I recommend setting a custom error like `setError({msg:"Failed to delete comment"})` and this will make it such that you can handle the errors dynamically and the messages will still be accurate